app/models/hackney/service_charge/letter.rb

```diff
@@ -87,36 +85,19 @@ def reorganise_address(letter_params)
         letter_params
       end
 
-      def validate_mandatory_fields(letter_params)
-        @errors = MANDATORY_LETTER_FIELDS
+      def validate_mandatory_fields(mandatory_fields, letter_params)
+        @errors = [] if @errors.nil?
```

Comment:
More idiomatic Ruby:

```suggestion
        @errors ||= []
```